assets/js/settings/blocks/constants.js

```diff
@@ -3,94 +3,64 @@
  */
 import { getSetting } from '@woocommerce/settings';
 
-export const CURRENT_USER_IS_ADMIN = getSetting( 'currentUserIsAdmin', false );
-export const REVIEW_RATINGS_ENABLED = getSetting(
-	'reviewRatingsEnabled',
-	true
-);
-export const SHOW_AVATARS = getSetting( 'showAvatars', true );
-export const MAX_COLUMNS = getSetting( 'max_columns', 6 );
-export const MIN_COLUMNS = getSetting( 'min_columns', 1 );
-export const DEFAULT_COLUMNS = getSetting( 'default_columns', 3 );
-export const MAX_ROWS = getSetting( 'max_rows', 6 );
-export const MIN_ROWS = getSetting( 'min_rows', 1 );
-export const DEFAULT_ROWS = getSetting( 'default_rows', 3 );
-export const MIN_HEIGHT = getSetting( 'min_height', 500 );
-export const DEFAULT_HEIGHT = getSetting( 'default_height', 500 );
-export const PLACEHOLDER_IMG_SRC = getSetting( 'placeholderImgSrc', '' );
-export const THUMBNAIL_SIZE = getSetting( 'thumbnail_size', 300 );
-export const IS_LARGE_CATALOG = getSetting( 'isLargeCatalog' );
-export const LIMIT_TAGS = getSetting( 'limitTags' );
-export const HAS_PRODUCTS = getSetting( 'hasProducts', true );
-export const HAS_TAGS = getSetting( 'hasTags', true );
-export const HOME_URL = getSetting( 'homeUrl', '' );
-export const COUPONS_ENABLED = getSetting( 'couponsEnabled', true );
-export const SHIPPING_ENABLED = getSetting( 'shippingEnabled', true );
-export const TAXES_ENABLED = getSetting( 'taxesEnabled', true );
-export const DISPLAY_ITEMIZED_TAXES = getSetting(
-	'displayItemizedTaxes',
-	false
-);
-export const HAS_DARK_EDITOR_STYLE_SUPPORT = getSetting(
-	'hasDarkEditorStyleSupport',
-	false
-);
-export const DISPLAY_SHOP_PRICES_INCLUDING_TAX = getSetting(
-	'displayShopPricesIncludingTax',
-	false
-);
-export const DISPLAY_CART_PRICES_INCLUDING_TAX = getSetting(
-	'displayCartPricesIncludingTax',
-	false
-);
-export const PRODUCT_COUNT = getSetting( 'productCount', 0 );
-export const ATTRIBUTES = getSetting( 'attributes', [] );
-export const IS_SHIPPING_CALCULATOR_ENABLED = getSetting(
-	'isShippingCalculatorEnabled',
-	true
-);
-export const IS_SHIPPING_COST_HIDDEN = getSetting(
-	'isShippingCostHidden',
-	false
-);
-export const WOOCOMMERCE_BLOCKS_PHASE = getSetting(
-	'woocommerceBlocksPhase',
-	1
-);
-export const WC_BLOCKS_ASSET_URL = getSetting( 'wcBlocksAssetUrl', '' );
-export const WC_BLOCKS_BUILD_URL = getSetting( 'wcBlocksBuildUrl', '' );
-export const SHIPPING_COUNTRIES = getSetting( 'shippingCountries', {} );
-export const ALLOWED_COUNTRIES = getSetting( 'allowedCountries', {} );
-export const SHIPPING_STATES = getSetting( 'shippingStates', {} );
-export const ALLOWED_STATES = getSetting( 'allowedStates', {} );
-export const SHIPPING_METHODS_EXIST = getSetting(
-	'shippingMethodsExist',
-	false
-);
+const blockConstants = getSetting( 'block_constants', {} );
```

Comment:
This is a breaking change, I know we're not using this publicly yet but it should be highlighted at least.
For example, I used `getSetting( 'displayCartPricesIncludingTax', false );` which is not longer available.
What's the reason for moving all data under `block_constants`?

Reply:
With the refactor each block tells the assets registry that it needs the constants. Until a block rendered happens the AssetsRegistery will not trigger the constants creation. This was the sole purpose of the refactor - no to trigger the wc_data_creation on every route. **And now with that in mind the reason for the `block_constant`:** since this is no longer global and each block does it is own registering we need a way to easily check if the data was already requested - so we will be able to avoid unnecessary multiple registers. This is managed by the AbstractBlock:

https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/3864/files#diff-d1b778e863fef808d0b34565afe0c8bc172cc3d9e3e99324b402a19577f1fbf1R333-R336

```PHP
// Enqueue common data.
if ( ! $this->asset_data_registry->exists( 'block_constants' ) ) {
	$this->asset_data_registry->add( 'block_constants', [ self::class, 'get_wc_block_data' ] );
}
```

We have only one key that we need to check. Any other dynamic approach way would require adjusting the AssetsDataRegistery framework for this special case of blocks common data which I think we should not do.

Reply:
I agree with Nadir here, this is a pretty significant breaking change and should be reconsidered. I'm not sure I agree with the solution in this refactor. I agree there needs to be a way for each block to register what data it needs (without triggering the error from duplicate keys) but I would prefer to see that happen at the block level. To avoid having to check if the data is available before registering (via `AssetsDataRegistry::exists`) - we could add a new helper method `addIfNotExists` or something like that which absorbs the necessary `exists` check and avoids registering duplicate data on the same key.

Then blocks that share data could potentially do the registration via a method on the abstract block class.

Some variation of the above should work and allows us to preserve back compat.